### ISSUE TITLE
command-reference.md in this repo?

### ISSUE THREAD

**Issue by MSP-Greg** [rubygems/rubygems#1865]
I would like to suggest a feature.

[command-reference.md](https://msp-greg.github.io/ruby_trunk/rubygems/file.command-reference.html)  (generated by code from [rubygems / guides](https://github.com/rubygems/guides)) is certainly a helpful/handy document.  Could it be added to this repo?

I'd be happy add a PR with the version I've got, but it's YARD specific...

**Comment by segiddins** [comment]
I think we want to keep it where it is, so that it wont become out-of-sync -- it's meant to be on the website (guides.rubygems.org)

**Comment by MSP-Greg** [comment]
My thought was by including the md file in ruby/ruby, it would be seen by anyone doc'ing ruby, and probably [ruby-doc.org](http://ruby-doc.org/).

[command_reference](http://guides.rubygems.org/command-reference) shows --

`This reference was automatically generated from RubyGems version 2.6.6.`

**Comment by segiddins** [comment]
If you want it to go into ruby/ruby, that's probably a conversation to take up with the Ruby maintainers?

**Comment by MSP-Greg** [comment]
Umm, where does ruby/ruby get it's updates from?  Here?

**Comment by segiddins** [comment]
As far as I know, they just pull in `lib` and `bin`

**Comment by drbrain** [comment]
tl;dr: I'm not convinced, maybe the problem is more fundamental?

Despite being a long-time promoter of reading documentation on your local machine, I don't believe that it is a thing people regularly do. When you have internet it is almost always more convenient to use search engines than the search in your local copy.

The page on guides.rubygems.org is sourced from `gem help`, so I'd prefer to make `gem help` better first.

We already have to close many pull requests on guides.rubygems.org that attempt to update either the specification reference or the command reference (the canonical source of each is in this repo).  I wouldn't feel comfortable committing the file here because it becomes a barrier to contribution. A PR to change that text must be made against the source strings embedded within the commands.

Adding the generated document to release tarballs would be OK, but if I were to implement it I would probably generate the file from a new option to `gem help` like `--all`.

This makes me wonder if there is a more fundamental problem like `gem help` is not helpful enough to people to intuitively find what they need.

**Comment by MSP-Greg** [comment]
@drbrain 

More fundamental?  Just kidding.  A concrete issue is that [command-reference](http://guides.rubygems.org/command-reference/) is v2.6.6.  For someone with access, the time that's been spent writing in this issue could have fixed that.

Maybe I'm a singularity, but I often look at doc on my tablet.  Hence, internet/browser access is a given.  That is the original reason that I started working on the code that's used to generate the doc site, which uses [command-reference.md](https://msp-greg.github.io/ruby_trunk/rubygems/file.command-reference.html), which is in my local ruby/ruby repo.

Conversely, I was working on a repo, and a person who's a good contributor to at least a few ruby gems mentioned an issue with 2.6.8 gem server.  So, there I was at guides, going 'what? 2.6.6?'

I originally posted this issue in ruby/ruby, because that's where I'd like the file, but someone said 'Third Party Issue/go upstream', so here I am.  ruby/ruby can't pull a file that isn't here.

> We already have to close many pull requests on guides.rubygems.org that attempt to update either the specification reference or the command reference (the canonical source of each is in this repo). I wouldn't feel comfortable committing the file here because it becomes a barrier to contribution. A PR to change that text must be made against the source strings embedded within the commands.

We all have to deal with people unaware of things, and I often am in that group (none of us knows everything...)  I don't think that's a good reason to keep a doc that many would find helpful out of the repo (or ruby/ruby).

Anyway, thanks...
